### PR TITLE
Feature/integration spec

### PR DESCRIPTION
**Pull request by Leboroz**
# Learning objectives
- Write integration tests with Capybara gem.

Project requirements
Use Capybara to write integration tests for each view in your project.

## Login page:
- 
- I can see the username and password inputs and the "Submit" button.
- When I click the submit button without filling in the username and the password, I get a detailed error.
- When I click the submit button after filling in the username and the password with incorrect data, I get a detailed error.
- When I click the submit button after filling in the username and the password with correct data, I am redirected to the root page.
## User index page:

- I can see the username of all other users.
- I can see the profile picture for each user.
- I can see the number of posts each user has written.
- When I click on a user, I am redirected to that user's show page.
## user show page:

- I can see the user's profile picture.
- I can see the user's username.
- I can see the number of posts the user has written.
- I can see the user's bio.
- I can see the user's first 3 posts.
- I can see a button that lets me view all of a user's posts.
- When I click a user's post, it redirects me to that post's show page.
- When I click to see all posts, it redirects me to the user's post's index page.
## User post index page:

- I can see the user's profile picture.
- I can see the user's username.
- I can see the number of posts the user has written.
- I can see a post's title.
- I can see some of the post's body.
- I can see the first comments on a post.
- I can see how many comments a post has.
- I can see how many likes a post has.
- I can see a section for pagination if there are more posts than fit on the view.
- When I click on a post, it redirects me to that post's show page.
## Post show page:

- I can see the post's title.
- I can see who wrote the post.
- I can see how many comments it has.
- I can see how many likes it has.
- I can see the post body.
- I can see the username of each commentor.
- I can see the comment each commentor left.

### CODE REVIEW

File: spec/system/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,100 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts', type: :system do
```
Comment (SajjadAhmad14) [inline spec/system/posts_spec.rb:3]: There are some tests failing. Please make sure all the tests pass.
fails

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Nicolas, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
README is professional. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: spec/system/device_spec.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    driven_by(:rack_test)
+  end
+
+  describe 'loads' do
+    before :each do
+      visit new_user_session_path
+    end
+
+    it 'the username and password inputs and the "Submit" button.' do
+      expect(page).to have_content 'Email'
```
Comment (Cathella) [inline spec/system/device_spec.rb:14]: ✅ Has featured test for login page 👍🏻. This is a good trial on writing first test. You can also use the suggestion below and find away to refactor your test
```ruby
describe 'the signin process', type: :feature do
     before :each do
         User.create(email: 'user1@gmail.com', password: 'password', name: 'User1')
     end

     it 'signs @user in' do
         visit '/users/sign_in'
         fill_in 'Email', with: 'user1@gmail.com'
         fill_in 'Password', with: 'password'
         click_button 'Log in'
         expect(current_path).to eq(root_path)
         expect(page).to have_text('Signed in successfully.')
     end

     it 'does not sign in with wrong info' do
         visit '/users/sign_in'
         fill_in 'Email', with: 'user1@gmail.com'
         fill_in 'Password', with: 'mypassword'
         click_button 'Log in'
         expect(page).to have_text('Invalid Email or password.')
     end
 end
```

**Review by Cathella — APPROVED** [review]
## 🔥 Status: Approved 🟢

### Hi @Leboroz 👋🏻, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💪🏻

✅ No linter errors.
✅ You used correct Github Flow.
✅ Professional README.
✅ Descriptive Pull request.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**